### DISCUSSION TITLE
Predicting what next?

### DISCUSSION THREAD

**Discussion by action2-jacksmiths** [TDAmeritrade/stumpy#927] [General]
Hello, 

I have just come across this project and find it very interesting. In your documentation you have mentioned this link for forecasting:
https://www.youtube.com/watch?v=KJNfm5m28y4
however, there is no further documentation on how to forecast? can you pls help guide

**Comment by seanlaw** [comment]
@action2-jacksmiths Thank you for your question and welcome to the STUMPY community. Can you please provide more detail on where we had mentioned the link above? Technically, STUMPY is NOT a forecasting tool. There are some things like [time series chains](https://stumpy.readthedocs.io/en/latest/Tutorial_Time_Series_Chains.html) that may seem similar to forecasting but I think you're better off using other tools for forecasting.

**Comment by action2-jacksmiths** [comment]
https://www.cs.ucr.edu/~eamonn/100_Time_Series_Data_Mining_Questions__with_Answers.pdf
page 17, 

![image](https://github.com/TDAmeritrade/stumpy/assets/93058740/6a6a1348-a407-4edc-aadc-d5d56513600f)

**Reply by seanlaw** [reply]
> In your documentation you have mentioned this link for forecasting: https://www.youtube.com/watch?v=KJNfm5m28y4
however, there is no further documentation on how to forecast? can you pls help guide

Unfortunately, that is NOT our documentation. What you are referencing is the website for the original academic researchers, which we are not affiliated with. STUMPY is purely supported by volunteers and our goal is to help reproduce the academic work. For questions related to forecasting, I recommend that you email the original authors directly.

**Reply by action2-jacksmiths** [reply]
sorry but this response is really confusing? on your github under readme you also link to the document? So i am confused!

**Reply by seanlaw** [reply]
Yes, our reference (https://www.cs.ucr.edu/~eamonn/100_Time_Series_Data_Mining_Questions__with_Answers.pdf) was to direct you to other things that could be done with matrix profiles in general and to redirect you to a primary source from where you can continue to follow up with the original academic research and work. We try to provide relevant resources for you to continue your exploration further. As you noticed in our README, the link was in reference to "matrix profile... can then be used for a variety of time series data mining tasks such as... and more". The reference was specifically to direct you to a list where others have done even MORE than what we are showing in STUMPY. However, it is up to you to dive deeper into the materials.

Note, we (STUMPY - unaffiliated with the original academic researchers) did not mention anything about the YouTube link that you has provided. Additionally, we did NOT write the PDF that is found in https://www.cs.ucr.edu/~eamonn/100_Time_Series_Data_Mining_Questions__with_Answers.pdf

**Comment by andydoc** [comment]
There is clearly a role in timeseries prediction. Unfortunately I do not have and can not afford computing resources to develop this, but my thesis is that:

- given a motif of length x commencing at T[t],
- predictability would require two submotifs of length y and length z, where len(x)=len(y)+len(z)
- the startpoint of x and y would be simultaneous at T[t], and that of z would be T[t+len(y)]
- motif z could then be predicted at T[t+len(y)]

x y and z could occur across multiple timeseries
this could be extended to more than 2 submotifs and it might also be interesting to explore the effect of various degrees of gap or overlap between the end of y and the beginning of z

This would be of potential interest in financial timeseries for example

**Reply by andydoc** [reply]
Adding to which, obviously the first submotif to identify would be z, since we want to find the predictor series y where it occurs before z with an acceptable degree of specificity

**Reply by action2-jacksmiths** [reply]
many thanks. Would you be able to give an example as the approach above sounds quite complex for me to implement?

**Reply by andydoc** [reply]
Sorry, as I said i have no resources to develop this or even explore the possibilities. My interest though is in trading, and I can see that given a profitable chart pattern (second sub-motif) (eg with a risk reward ration of 1:3 given fixed price movements defined as a % of the current price where the profit price is hit before the stop-loss price) we could identify a predictor (first sub-motif) pattern so that overall entering a trade at the prediction point would be profitable.
This could be filtered further by time of day, and other series added so that open high and low as well as close, along with volume were considered.
The price (P) timeseries should be rendered stationary by using return (P[t]-P[t-1])/P[t-1]) of the OHLC series, and, along with volume, normalised. Volume could be rendered stationary if it appears necessary.

**Reply by andydoc** [reply]
https://medium.com/@denis.vivdenko.freelancer/how-i-built-an-algorithm-that-finds-patterns-in-stock-prices-unsupervised-learning-c5dbafdc4e26

Looks interesting...
